### ISSUE TITLE
Renaming on windows

### ISSUE THREAD

**Issue by Ewan-Walker** [radovskyb/watcher#16]
It would seem that on windows there is no rename event but rather the following:

- rename file/folder
- CREATE event (new name)
- REMOVE event (old name)

So there is never a RENAME event actually ever firing.

**Comment by radovskyb** [comment]
Hey @Ewan-Walker, 

It seems that this is only now a problem since I rewrote Watcher recently.
I am currently thinking it might be an issue with os.SameFile not working consistently on Windows but I will have to do some investigations.

Thanks for raising this issue. Will have a deeper look in my spare time!

**Comment by radovskyb** [comment]
Hey again @Ewan-Walker, 

I've implemented a small fix for now, so if you re run `go get -u github.com/radovskyb/watcher/...` it should be working for windows now.

Like I thought, the issue is that `os.SameFile` doesn't work consistently across cross platforms, so I'll have to look into that too.

Either way, this fix should work for now :)

I've also noticed my tests fail on windows for chmod events, so I'll have to look into that as well.
